### PR TITLE
Add cargo make tasks for packaging/publishing

### PR DESCRIPTION
**Pull request by russcam**
This commit adds cargo make tasks for packaging and publishing, including generating release notes

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -21,6 +23,12 @@ private = true
 condition = { env = { "TEST_SUITE" = "xpack" }, env_not_set = ["ELASTICSEARCH_URL"] }
 env = { "ELASTICSEARCH_URL" = "https://elastic:changeme@localhost:9200" }
 
+[tasks.set-elasticsearch-crate-env]
```
Comment (swallez) [inline Makefile.toml:26]: This task doesn't seem to be used. Leftover from a previous iteration?

File: Makefile.toml [hunk 2]
```diff
@@ -139,6 +160,57 @@ description = "Generates Elasticsearch client from REST API specs"
...
+script = [
+"""
+pushd ./../clients-team/scripts/release-notes-generator
+docker build --file ./Dockerfile --tag clients-team/release_notes_generator .
+popd
+docker run -v "${CARGO_MAKE_WORKING_DIRECTORY}/.ci/release/config.yml:/usr/src/release_notes_generator/config.yml" --rm clients-team/release_notes_generator -o ${OLD_VERSION} -n ${NEW_VERSION}
+"""
+]
+
+[tasks.generate-release-notes.windows]
```
Comment (swallez) [inline Makefile.toml:185]: Could the use of `@shell` or Duckscript avoid a platform-specific implementation of this task?

[shell2batch](https://github.com/sagiegurari/shell2batch) doesn't support `pushd` but we can replace it with `cd`

File: Makefile.toml (hunk 1 above)
Comment (russcam) [reply Makefile.toml:26]: Yes, good catch! Will remove

File: Makefile.toml (hunk 2 above)
Comment (russcam) [reply Makefile.toml]: ++ using `@shell` works

File: Makefile.toml [hunk 3]
```diff
@@ -81,6 +89,19 @@ condition = { env_set = [ "CI" ] }
...
+[tasks.publish-impl]
+description = "Runs the cargo publish command."
+category = "Elasticsearch"
+private = true
+script_runner = "@shell"
+script = [
+"""
+pushd elasticsearch
+cargo publish ${CARGO_MAKE_CARGO_PUBLISH_FLAGS}
+popd
```
[the remark this answers is not in the document]
Comment (russcam) [reply Makefile.toml]: ++ simplified using `@duckscript`

File: Makefile.toml (hunk 1 above)
Comment (russcam) [reply Makefile.toml]: removed

**Review by swallez — APPROVED** [review]
LGTM!

**Pull request merged**